src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -258,36 +274,96 @@ where
 
         // Install collection state for each bound source.
         for (id, (desc, since)) in bindings {
-            let ts_binding_collection = self
-                .state
-                .stash
-                .collection::<PartitionId, ()>(&format!("timestamp-bindings-{id}"))?;
+            let (command, collection_state) = if let SourceConnector::External {
+                connector: ExternalSourceConnector::Persist(ref persist_connector),
+                ..
+            } = desc.connector
```

Comment:
I have a bunch of thoughts here that might need to be follow-on work. Mostly of the form "we want to find a way to have the controller contain less of this sort of logic itself, and instead shell out to implementations elsewhere". Otherwise the controllers will become the new `coord.rs`.

Reply:
I maybe addressed this by moving the code that decides to create a read handle to the source connector.